server/app/services/applicant/question/NameQuestion.java

```diff
@@ -72,8 +82,15 @@ public Optional<String> getMiddleNameValue() {
       return middleNameValue;
     }
 
-    middleNameValue = applicantQuestion.getApplicantData().readString(getMiddleNamePath());
+    ApplicantData applicantData = applicantQuestion.getApplicantData();
+    middleNameValue = applicantData.readString(getMiddleNamePath());
```

Comment:
Please check if the ReadString()  also handles cases where no path exists. Like few applicants may not have a middle name.

Reply:
It is not an issue. Optional<String> saved the day :)